src/commands/open.js

```diff
@@ -1,25 +1,42 @@
 const log = require('@dhis2/cli-helpers-engine').reporter
-
+const concurrently = require('concurrently')
 const { cypress } = require('../tools/cypress.js')
+const { appStart, waitOn } = require('./common/sharedCLIOptions.js')
 
 exports.command = 'open'
 
 exports.aliases = ['o']
 
 exports.desc = 'Open Cypress UI'
 
-exports.builder = {}
+exports.builder = yargs =>
+    yargs.option('appStart', appStart).option('waitOn', waitOn)
 
 exports.handler = argv => {
     log.info('d2-utils-cypress > open')
-
-    const { port, browser } = argv
+    const { appStart, port, browser, waitOn } = argv
 
     const opts = {
         mode: 'open',
         browser,
         port,
     }
 
-    cypress(opts)
+    if (!appStart) {
+        log.info('"appStart" empty. Running cypress directly')
+
+        cypress(opts)
+    } else {
+        const cypressCommand = cypress({ ...opts, exec: false })
+        const waitOnCommand = `npx --no-install wait-on ${waitOn}`
```

Comment:
This works great in apps, but when used through the global install of d2 (i.e. `d2 utils cypress`) this doesn't work for the same reasons it doesn't in `d2 style`.

I've been working on an alternative style that shouldn't get confused when e.g. `d2` is installed globally using Yarn vs. NPM (they have different `node_modules` globally, and `npx` looks in the NPM one and not Yarn one).

There's also the case of not hoisting dependencies (not talking about de-duping) that are _only_ used in e.g. `d2-style` which this "inside out" approach allows to work. Needs more testing and bug fixing but it's in `@dhis2/cli-style@alpha` right now: https://github.com/dhis2/cli-style/commit/fd620c56ca8798ab0e84366d97cdbd0d0fdaba03

If it works out, maybe we should move the `cli-style/src/utils/run.js` file to e.g. `cli-helpers-engine` and use that as the primary process runner.

Reply:
What I'm saying is: use `npx` for now, and be aware that there are some gotchas that we want to resolve, and there are experiments for that in `dhis2/cli-style` going on.

Reply:
Good to know. So if I get it right `d2 utils cypress open` won't work when using the globally installed cli utility. Never thought about that use-case before :sweat_smile:

Reply:
I think `npx` is probably OK here for the moment because `wait-on` doesn't need to find other dependencies (like `eslint` does when resolving plugins from the .eslintrc file).  So should be fine for now but worth moving to a better process runner when we have it.